themes/custom/az_barrio/az_barrio.theme

```diff
@@ -11,6 +11,90 @@ include_once drupal_get_path('theme', 'az_barrio') . '/includes/common.inc';
 use Drupal\Core\Link;
 use Drupal\Core\Render\Markup;
 use Drupal\Component\Utility\Html;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element;
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ */
+function az_barrio_theme_suggestions_form_element_alter(&$suggestions, $variables) {
+  $suggestions = ['form_element__type__' . $variables['element']['#type']];
+  if (isset($variables['element']['#form_id'])) {
+    $suggestions[] = 'form_element__form_id__' . $variables['element']['#form_id'];
+    $suggestions[] = 'form_element__' . $variables['element']['#form_id'] . '__' . $variables['element']['#type'];
+  }
+}
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ */
+function az_barrio_theme_suggestions_form_element_label_alter(&$suggestions, $variables) {
+  if (isset($variables['element']['#form_element_type'])) {
+    $suggestions[] = 'form_element_label__type__' . $variables['element']['#form_element_type'];
+  }
+  if (isset($variables['element']['#form_id'])) {
+    $suggestions[] = 'form_element_label__form-id__' . $variables['element']['#form_id'];
+  }
+  if (isset($variables['element']['#form_element_type']) && isset($variables['element']['#form_id'])) {
+    $suggestions[] = 'form_element_label__' . $variables['element']['#form_id'] . '__' . $variables['element']['#form_element_type'];
+  }
+}
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ */
+function az_barrio_theme_suggestions_input_alter(&$suggestions, $variables) {
+  if (isset($variables['element']['#type'])) {
+    $suggestions[] = 'input__' . $variables['element']['#type'];
+  }
+  if (isset($variables['element']['#form_id'])) {
+    $suggestions[] = 'input__' . $variables['element']['#form_id'];
+  }
+  if (isset($variables['element']['#type']) && isset($variables['element']['#form_id'])) {
+    $suggestions[] = 'input__' . $variables['element']['#form_id'] . '__' . $variables['element']['#type'];
+  }
+}
+
+/**
+ * Implements hook_preprocess_form_element().
+ */
+function az_barrio_preprocess_form_element(array &$variables) {
+  if (!isset($variables['label'])) {
+    return;
+  }
+  if (isset($variables['element']['#form_id'])) {
+    $variables['label']['#form_id'] = $variables['element']['#form_id'];
+  }
+  if (isset($variables['element']['#type'])) {
+    $variables['label']['#form_element_type'] = $variables['element']['#type'];
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function az_barrio_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  az_barrio_attach_form_id($form, $form_id);
```

Comment:
Would it be possible to limit the scope of this `hook_form_alter()` implementation?  I'm not sure we really want to alter _all_ forms (and form elements) like this.